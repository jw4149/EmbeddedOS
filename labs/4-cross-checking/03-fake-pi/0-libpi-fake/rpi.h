#ifndef __RPI_H__
#define __RPI_H__
// simplest way to rewrap --- just define replacements in header --- don't
// even need .c
//
// easy, but two downsides: 
//   1. we are polluting the pi program namespace with system includes.
//   2. we aren't using the raw rpi.h so can get different behavior.
#include <stdio.h>
#include <stdlib.h>

// redefine printk as fprintf using variadic macros.
#define printk(args...) fprintf(stderr, args)

static void inline clean_reboot(void) {
    printk("DONE!!!\n");
    exit(0);
}

void notmain(void);
#endif
