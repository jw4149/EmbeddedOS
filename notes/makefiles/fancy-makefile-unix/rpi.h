#include <stdio.h>

#define printk printf
#define debug printf
