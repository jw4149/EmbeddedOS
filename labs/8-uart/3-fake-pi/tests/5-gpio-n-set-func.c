#include "rpi.h"
void notmain(void) {
    for(int i = 0; i < 7; i++)
        gpio_set_function(20, i);
}
