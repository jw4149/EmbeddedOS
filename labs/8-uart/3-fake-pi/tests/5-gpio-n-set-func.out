TRACE:0: calling pi code
TRACE:1: GET32(0x20200008) = 0x66334873
TRACE:2: PUT32(0x20200008) = 0x66334870
TRACE:3: GET32(0x20200008) = 0x66334870
TRACE:4: PUT32(0x20200008) = 0x66334871
TRACE:5: GET32(0x20200008) = 0x66334871
TRACE:6: PUT32(0x20200008) = 0x66334872
TRACE:7: GET32(0x20200008) = 0x66334872
TRACE:8: PUT32(0x20200008) = 0x66334873
TRACE:9: GET32(0x20200008) = 0x66334873
TRACE:10: PUT32(0x20200008) = 0x66334874
TRACE:11: GET32(0x20200008) = 0x66334874
TRACE:12: PUT32(0x20200008) = 0x66334875
TRACE:13: GET32(0x20200008) = 0x66334875
TRACE:14: PUT32(0x20200008) = 0x66334876
TRACE: pi exited cleanly: 7 calls to random
